File: src/main/java/awkMULE/GameState.java
package awkMULE;

import java.util.ArrayList;

public class GameState {

	private static GameState instance;
	private int numPlayers;

	private ArrayList<Player> players;

	private int currentPlayerIndex;

	public GameState(int numPlayers) {
		players = new ArrayList<>();
		this.numPlayers = numPlayers;
		currentPlayerIndex = 0;
	}

	public ArrayList<Player> getPlayers() {
		return players;
	}

	public int getNumPlayers() {
		return numPlayers;
	}

	public Player getCurrentPlayer() {
		return players.get(currentPlayerIndex);
	}

	public void setNumPlayers(int numPlayers) {
		this.numPlayers = numPlayers;
	}

	public void addPlayer(Player player) throws GameStateConfigException {
		for (Player p: players) {
			if (player.getName().equals(p.getName())) {
				throw new GameStateConfigException("duplicate name");
			}
		}
		players.add(player);
	}

	public static GameState getInstance() {
		if (GameState.instance == null) {
			GameState.instance = new GameState(0);
		}
		return GameState.instance;
	}
}

File: web/current/web/GameState.dart
import 'Player.dart';
import 'GameMap.dart';
import 'Store.dart';

class GameState {
    Store store = new Store();
    List<Player> players = [];
    int num_players = 0;
    int round = 0;
    GameMap map;
    int currentPlayerIndex = 0;

    GameState(this.num_players, this.map);

    Player get currentPlayer {
        if (currentPlayerIndex >= players.length) {
            print("player index out of bounds");
            return null;
        }
        return players[currentPlayerIndex];
    }
}